libbeat/cfgfile/cfgfile.go

```diff
@@ -62,3 +66,27 @@ func Read(out interface{}, path string) error {
 func IsTestConfig() bool {
 	return *testConfig
 }
+
+// expandEnv replaces ${var} or $var in config according to the values of the
+// current environment variables. The replacement is case-sensitive. References
+// to undefined variables are replaced by the empty string. A default value
+// can be given by using the form ${var:default value}.
+func expandEnv(config []byte) []byte {
```

Comment:
Is there a reason we can't use the function provided by golang? https://golang.org/pkg/os/#ExpandEnv